src/ride/ride.c

```diff
@@ -995,7 +995,9 @@ void ride_remove_peeps(int rideIndex)
 			exitZ = ride->station_heights[stationIndex];
 			mapElement = ride_get_station_exit_element(ride, exitX, exitY, exitZ);
 
-			exitDirection = mapElement->type & 3;
+			if (mapElement != NULL) {
+				exitDirection = mapElement->type & 3;
+			}
```

Comment:
Yeah, I just realised that and issued a fix by clamping `exitDirection`
regardless of it being set or not inside that `if`.

On Sun, Jan 3, 2016 at 11:56 AM, Duncan <notifications@github.com> wrote:

> In src/ride/ride.c
> <https://github.com/OpenRCT2/OpenRCT2/pull/2629#discussion_r48691899>:
>
> > @@ -995,7 +995,9 @@ void ride_remove_peeps(int rideIndex)
> >  			exitZ = ride->station_heights[stationIndex];
> >  			mapElement = ride_get_station_exit_element(ride, exitX, exitY, exitZ);
> >
> > -			exitDirection = mapElement->type & 3;
> > +			if (mapElement != NULL) {
> > +				exitDirection = mapElement->type & 3;
> > +			}
>
> This is worse. Your now accessing completely invalid memory and silently
> continuing since exitDirection is set to 255. I would instead set it to 0.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenRCT2/OpenRCT2/pull/2629/files#r48691899>.
>



-- 
Michal Janiszewski